components/motor/gpio/encoded.go

```diff
@@ -363,28 +364,18 @@ func (m *EncodedMotor) ResetZeroPosition(ctx context.Context, offset float64, ex
 	return nil
 }
 
-// report position in ticks.
-func (m *EncodedMotor) position(ctx context.Context, extra map[string]interface{}) (float64, error) {
-	ticks, _, err := m.encoder.Position(ctx, encoder.PositionTypeTicks, extra)
-	if err != nil {
-		return 0, err
-	}
-	m.mu.RLock()
-	defer m.mu.RUnlock()
-	pos := ticks + m.offsetInTicks
-	return pos, nil
-}
-
 // Position reports the position of the motor based on its encoder. If it's not supported, the returned
 // data is undefined. The unit returned is the number of revolutions which is intended to be fed
 // back into calls of GoFor.
 func (m *EncodedMotor) Position(ctx context.Context, extra map[string]interface{}) (float64, error) {
-	ticks, err := m.position(ctx, extra)
+	ticks, _, err := m.encoder.Position(ctx, encoder.PositionTypeTicks, extra)
```

Comment:
Should we check position type here with helper code? If the encoder dependency rebuilt randomly for one that can be both it's a nice check.

Reply:
yeah this should validate teh encoder type, will update